CodeConverter/CSharp/PerScopeState.cs

```diff
@@ -102,18 +113,41 @@ public async Task<SyntaxList<StatementSyntax>> CreateLocalsAsync(VBasic.VisualBa
             return CS.SyntaxFactory.List(statementsWithUpdatedIds);
         }
 
-        public static IEnumerable<StatementSyntax> ReplaceNames(IEnumerable<StatementSyntax> csNodes, Dictionary<string, string> newNames)
+
+        public async Task<SyntaxList<MemberDeclarationSyntax>> CreateVbStaticFieldsAsync(VBasic.VisualBasicSyntaxNode vbNode, IEnumerable<MemberDeclarationSyntax> csNodes, HashSet<string> generatedNames, SemanticModel semanticModel)
+        {
+            var declarations = new List<FieldDeclarationSyntax>();
+
+            var fieldInfo = GetFields();
+            var newNames = fieldInfo.ToDictionary(f => f.OriginalVariableName, f =>
+                NameGenerator.GetUniqueVariableNameInScope(semanticModel, generatedNames, vbNode, f.FieldName)
+            );
+            foreach (var field in fieldInfo) {
+                var decl = CommonConversions.CreateVariableDeclarationAndAssignment(newNames[field.OriginalVariableName],
+                    field.Initializer, field.Type);
+                var modifiers = new List<SyntaxToken> { CS.SyntaxFactory.Token(SyntaxKind.PrivateKeyword) };
+                if (field.IsStatic) {
+                    modifiers.Add(CS.SyntaxFactory.Token(SyntaxKind.StaticKeyword));
+                }
```

Comment:
Would be good to add a test for the case where the containing type is a module - I think you also need to check here for that case and make the field static